src/ks_thread.c

```diff
@@ -77,7 +77,16 @@ KS_DECLARE(ks_pid_t) ks_thread_self_id(void)
 #elif KS_PLAT_LIN
 	return syscall(SYS_gettid);
 #else
-	return pthread_self();
+    uint64_t tid;
+    int r = pthread_threadid_np(NULL, &tid);
+    if (!r) {
+        return tid;
+    } else {
+        ks_log(KS_LOG_CRIT, "pthread_threadid_np error, return %d", r);
+        ks_log(KS_LOG_CRIT, "BACKTRACE:");
+        ks_debug_dump_backtrace();
+        abort();
+    }
```

Comment:
The surrounding code uses tabs while this addition uses spaces for indentation.

Reply:
done.